src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -97,7 +103,7 @@ private object GetFirstOrDefaultValue(OptionKey optionKey, IEnumerable<RoamingPr
 
                 RecordObservedValueToWatchForChanges(optionKey, storageKey);
 
-                if (_settingManager.TryGetValue(storageKey, out object value) == GetValueResult.Success)
+                if (_serviceInitializer.GetService(CancellationToken.None).TryGetValue(storageKey, out object value) == GetValueResult.Success)
```

Comment:
this worries me from a perf perspective.  it seems like we'll just be shuffling this cost around.  Why not either take a real cancellation token, and pass it here?  Or use hte GetServiceAsync call and make this method async?

Reply:
work has to be done at some point, we don't want to do it at package initialization, so it is now moved to first place service is used.

setting manager is a bit better since service is free-thread service, unlike other native services which require UI thread so, only thing we can do is moving to GetServiceAsync from GetService.

but even for setting manager, some of our API is already exposed as synchronous API (Workspace.Options) not possible to move to Async without breaking changes.

if we decide to take breaking changes, we can simply change service.GetService to GetServiceAsync, until then GetService. not best but still this will no longer run from package initialization.

Reply:
> so it is now moved to first place service is used.

Ok.  But it still worries me that we're not passing a real cancellation token.  If this can be expensive (which is presumably why we're moving it off of package initialization), then why not make it cancellable or async?  Passing CancellationToken.None to stuff that is expensive is really just shifting the costs around.

Reply:
Keep in mind that the `CancellationToken` is not even used for querying for the service. So if the service is expensive, a CT won't help break out. The `ServiceInitializer` class only uses the CT while waiting for @heejaechang's the semaphore. But `GetService` and `GetServiceAsync` don't take a CT so that can't be helped.

What you could do is add a `.WithCancellation(CT)` extension method to what you're awaiting for, which doesn't cancel the original task, but it does allow the awaiter to resume and throw `OperationCanceledException` immediately upon cancellation.